pkg/prometheus/promcfg_test.go

```diff
@@ -8837,3 +8886,102 @@ scrape_configs:
 
 	}
 }
+
+func TestTracingConfig(t *testing.T) {
+	samplingTwo := resource.MustParse("0.5")
+	testCases := []struct {
+		tracingConfig  *monitoringv1.PrometheusTracingConfig
+		name           string
+		expectedConfig string
+		expectedErr    bool
+	}{
+		{
+			name: "Config without sampling and client type",
+			tracingConfig: &monitoringv1.PrometheusTracingConfig{
+				Endpoint: "https://otel-collector.default.svc.local:3333",
+			},
+
+			expectedConfig: `global:
+  evaluation_interval: 30s
+  scrape_interval: 30s
+  external_labels:
+    prometheus: default/test
+    prometheus_replica: $(POD_NAME)
+scrape_configs: []
+tracing:
+  endpoint: https://otel-collector.default.svc.local:3333
+  insecure: null
+  headers: {}
+  compression: null
+  timeout: null
```

Comment:
Looks strange to pass down null values in the config... maybe we should only write it down if different from nil?

what do you think

Reply:
I'm good with that!
But we came to the first design we have, many null checkers on our side.

Reply:
@ArthurSens don, can you check?